integrations/integrations.php

```diff
@@ -13,49 +18,72 @@ class Integrations {
 	/**
 	 * Collection of registered integrations.
 	 *
-	 * @var array<Integration>
+	 * @var array<string,Integration>
 	 */
 	private array $integrations = [];
 
 	/**
 	 * Registers an integration.
 	 *
-	 * @param string             $slug            A unique identifier for the integration.
-	 * @param string|Integration $class_or_object Fully-qualified class or instantiated Integration object.
+	 * @param Integration $integration Instantiated integration object.
+	 *
+	 * @throws InvalidArgumentException Exception if invalid argument are passed.
 	 *
 	 * @private
 	 */
-	public function register( string $slug, $class_or_object ): void {
-		if ( isset( $this->integrations[ $slug ] ) ) {
+	public function register( $integration ): void {
+		if ( ! is_subclass_of( $integration, Integration::class ) ) {
+			throw new InvalidArgumentException( sprintf( 'Integration class "%s" must extend %s.', get_class( $integration ), Integration::class ) );
+		}
+
+		$slug = $integration->get_slug();
+
+		if ( null !== $this->get_integration( $slug ) ) {
 			throw new InvalidArgumentException( sprintf( 'Integration with slug "%s" is already registered.', $slug ) );
 		}
 
-		if ( is_object( $class_or_object ) ) {
-			$integration_object = $class_or_object;
-		} else {
-			if ( ! class_exists( $class_or_object ) ) {
-				throw new InvalidArgumentException( sprintf( 'Integration class "%s" does not exist.', $class_or_object ) );
+		$this->integrations[ $slug ] = $integration;
+	}
+
+	/**
+	 * Returns a registered integration for a key, or null if not found.
+	 *
+	 * @param string $slug A unique identifier for the integration.
+	 */
+	private function get_integration( string $slug ): ?Integration {
+		return $this->integrations[ $slug ] ?? null;
+	}
+
+	/**
+	 * Activates integrations based on the configuration provided by VIP
+	 * (only if not already activated by customer).
+	 *
+	 * @return void
+	 */
+	public function activate_integrations_via_vip_config() {
```

Comment:
What do you think about renaming this to `activate_platform_managed_integrations` or even just `activate_managed_integrations`?

Reply:
Renamed to `activate_platform_integrations`